android/runtime/v8/src/native/TypeConverter.cpp

```diff
@@ -606,7 +606,7 @@ jobject TypeConverter::jsValueToJavaObject(v8::Isolate* isolate, v8::Local<v8::V
 
 jobject TypeConverter::jsValueToJavaObject(v8::Isolate* isolate, JNIEnv *env, v8::Local<v8::Value> jsValue, bool *isNew)
 {
-	if (jsValue->IsNumber()) {
+	if (jsValue->IsNumberObject()) {
```

Comment:
So clearly this is the correct thing to do, otherwise '1' returns true for IsNumber() leading to the bug we saw. However, are we certain this fix won't actually break other code? I'm guessing it shouldn't, but our Java code isn't consistent about enforcing HashMap\<String, Object\>(and besides, as seen here, generics are more a compile time suggestion and the mix of native code can get around it).